controller_interface/src/controller_interface.cpp

```diff
@@ -25,30 +26,105 @@ namespace controller_interface
 return_type
 ControllerInterface::init(const std::string & controller_name)
 {
-  lifecycle_node_ = std::make_shared<rclcpp_lifecycle::LifecycleNode>(
+  node_ = std::make_shared<rclcpp::Node>(
     controller_name,
     rclcpp::NodeOptions().allow_undeclared_parameters(true).
     automatically_declare_parameters_from_overrides(true));
+  return return_type::SUCCESS;
+}
 
-  lifecycle_node_->register_on_configure(
-    std::bind(&ControllerInterface::on_configure, this, std::placeholders::_1));
-
-  lifecycle_node_->register_on_cleanup(
-    std::bind(&ControllerInterface::on_cleanup, this, std::placeholders::_1));
-
-  lifecycle_node_->register_on_activate(
-    std::bind(&ControllerInterface::on_activate, this, std::placeholders::_1));
-
-  lifecycle_node_->register_on_deactivate(
-    std::bind(&ControllerInterface::on_deactivate, this, std::placeholders::_1));
+const rclcpp_lifecycle::State & ControllerInterface::configure()
```

Comment:
This looks good. Although I would recommend to create an additional header emilating Lifecycle Bode, so that controller_interface.cpp is changed minimally

Reply:
This is code, not a sacred text. I think it's ok to add these here with the understanding that we remove it when needed. It's even somewhat simpler to work with if you are a newcomer as it's all presented in a single file.

Reply:
I'd go with this and move forward with the release.

We should  open an issue for the no services feature, otherwise we'll be stuck with this forever.